presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1064,7 +1065,31 @@ public void alterTable(HiveIdentity identity, String databaseName, String tableN
                     .stopOnIllegalExceptions()
                     .run("alterTable", stats.getAlterTable().wrap(() -> {
                         try (ThriftMetastoreClient client = createMetastoreClient(identity)) {
-                            client.alterTable(databaseName, tableName, table);
+                            Map<String, String> tableParameters = table.getParameters();
+                            String statsParam = "DO_NOT_UPDATE_STATS";
+                            if ("true".equals(tableParameters.get(statsParam))) {
```

Comment:
Does it have to be conditional?

I don't think there is a situation where I would want metastore to collect the basic stats (we update them for managed tables, and for external tables you are now disabling them)

Reply:
I was trying to minimize the number of code paths affected by this change, because I didn't fully understand potential consequences.

Now that you mention it, we don't need the stats collected by Hive anyway, so it would make sense to make this unconditional.